src/js/store/middleware/feature/form.js

```diff
@@ -226,6 +265,46 @@ const handleClear = (store, next) => {
   ])
 }
 
+const handlePersistInLocalStorage = (store) => {
+  const state = store.getState()
+  const key = localStorageKey(state)
+  const storageState = {
+    schema: state.form.schema,
+    fields: state.form.fields,
+  }
+  window.localStorage.setItem(key, JSON.stringify(storageState))
```

Comment:
In response to `PERSIST_IN_LOCAL_STORAGE` actions (triggered in response to `INPUT_FIELD` actions), we store both the form schema and the current field values in local storage. We need the field values to restore the user's work, and we need the schema to ensure that the schema has not changed since the user input the stored data.

Reply:
We also store the current state of the UI to keep track of whether the user has loaded an index or not. If the user accepts local data on a form with an index we essentially need to spoof not only the field inputs but also the "index submission" step to let the app know that it should render the full form and not just the index fields.